RFC-0001-Agile Coretime.md

```diff
@@ -0,0 +1,298 @@
+
+### Broker-chain
+
+The Broker-chain is a new system parachain. It has the responsibility of providing the Relay-chain via UMP with scheduling information of:
+
+- The number of cores which should be made available during the next session.
+- Which cores should be allocated to which para IDs.
+
+Any cores left unallocated are assumed by the Broker-chain to be used for Instantaneous Coretime sales.
+
+### Parameters
+
+This proposal includes a number of parameters which need not necessarily be fixed. They are stated here along with a value, either *suggested* in which case it is non-binding and the proposal should not be judged on the value since the governance mechanism of Polkadot is expected to specify/maintain it; or *specified* in which case the proposal should be judged on the merit of the value as-is.
+
+| Name                | Value                        |
+| ------------------- | ---------------------------- |
+| `BULK_PERIOD`       | `28 * DAYS`                  |
+| `TIMESLICE`         | `10 * MINUTES`               |
+| `BULK_TARGET`       | `30`                         |
+| `BULK_LIMIT`        | `45`                         |
+| `LEADIN_PERIOD`     | `14 * DAYS`                  |
+| `RENEWAL_PRICE_CAP` | `Perbill::from_percent(2)`   |
+
+### Bulk Sales
+
+There is a periodic sale of Coretime happening at a period of `BULK_PERIOD`.  A number of instances of Coretime spanning from `LEADIN_PERIOD` in the future to `LEADIN_PERIOD + BULK_PERIOD` in the future of a single Polkadot UC Core is offered by the Polkadot System at a fixed price.
+
+These instances which are owned by purchaser are called *regions*. This sale happens on the Broker-chain. Regions are quantized into atomic periods of `TIMESLICE`, into which `BULK_PERIOD` divides a whole number of times. The `Timeslice` type is a `u16` which can be multiplied by `TIMESLICE` to give a `BlockNumber` value indicating the same period in times of (Relay-chain) blocks.
+
+The Broker-chain aims to sell some `BULK_TARGET` of Cores, up to some `BULK_LIMIT`. It makes this sale in a single batch `LEADIN_PERIOD` prior to the beginning of the period being sold. The Broker chain publishes a price for this batch of sales for the `BULK_TARGET` period prior to the sale execution.
+
+Accounts may call a `purchase` function with the appropriate funds in place to have their funds reserved and signal an intention to purchase Bulk Coretime for the forthcoming period. One account may have only one pending purchase. Any number of accounts may attempt to `purchase` Bulk Coretime for the forthcoming period, but the order is recorded.
+
+If there are more purchase requests than available cores for purchase in this period, then requests are processed in incoming order. Any additional purchase orders are carried over but marked as such. A purchase is only cancellable if it was carried over.
```

Comment:
This might be a potential MEV vector in case where bulk coretime becomes in such high demand that more orders are submitted within the first possible block than can be fulfilled and the validator decides the order in which to accept them.

Reply:
True, but month to month this should not be much of an issue as the price gets regulated properly.

Reply:
This has been modified to a hybrid auction system which should reduce the potential for MEV somewhat.